libs/SmartSync/src/com/salesforce/androidsdk/smartsync/manager/SyncManager.java

```diff
@@ -354,6 +354,9 @@ private void updateSync(SyncState sync, SyncState.Status status, int progress, S
                         }
                         attributes.put("syncId", sync.getId());
                         attributes.put("syncTarget", sync.getTarget().getClass().getName());
+                        if (sync.getEndTime() > 0 && sync.getStartTime() > 0) {
+                            attributes.put("runTime", sync.getEndTime() - sync.getStartTime());
```

Comment:
We shouldn't put this in `attributes`. The `attributes` part of the payload is for free-form JSON that doesn't have well-defined fields in the schema or for additional information. In this case, there's a `duration` field in the schema that is automatically extracted in `Splunk`, making it easy to look up. The library has 2 fields - [startTime](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/model/InstrumentationEvent.java#L66) and [endTime](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/model/InstrumentationEvent.java#L67). We automatically compute `duration` and put it in the payload [here](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceAnalytics/src/com/salesforce/androidsdk/analytics/transform/AILTNTransform.java#L114), if `endTime` is available, since `endTime` is an optional field. For this case, I suggest we tweak the implementation of [this](https://github.com/forcedotcom/SalesforceMobileSDK-Android/blob/master/libs/SalesforceSDK/src/com/salesforce/androidsdk/analytics/EventBuilderHelper.java#L107) helper method to extract `startTime` and `endTime` from the `attributes` payload and put them in their respective fields of the event if they exist. So this line would also change to put `startTime` and `endTime` in `attributes`.